src/main/java/com/andrey/currencyconverter/app/ApplicationFlow.java

```diff
@@ -0,0 +1,10 @@
+package com.andrey.currencyconverter.app;
+
+import com.andrey.currencyconverter.controllers.Controller;
+import com.andrey.currencyconverter.view.UserInterface;
+
+public interface ApplicationFlow {
+    void startFlow(UserInterface userInterface,
```

Comment:
Зачем нам передавать эти объекты каждый раз при вызове? Почему не заинжектил через конструктор?

Reply:
Не разобрался как обязать передавать в конструктор все параметры. 
Был вариант с абстрактным классом с конструктором.
```suggestion
    public abstruct class AbstructApplicationFlow implements ApplicationFlow {
               public AbstructApplicationFlow (UI, Repo, Controlr) {
               
               }
    }
```


У меня и ранее была проблема с пониманием что куда инжектить.
1. Создать целевой объект Reposotiry
2. Создать Controller, приняв Repo
3. Создать UI, приняв Controller
Или
Думал вынести в abstract factory и создавать семейством. Однако тогда нет возможность отдельно заменить слой репо, например, и т.д.

В результате вынес в метод, так как думал что передача единовременно при старте - не критично. 
Другая реализация - другой flow, другие параметры метода.

Reply:
В интерфейса и не будет конструктора, просто убери из метоад, а инжекти уже в классе реализации в конструкторе